### PR TITLE
FEATURE: Highlight code blocks for chat messages

### PR DESCRIPTION
**Pull request by martin-brennan**
This commit adds highlightJS decoration for chat messages,
so now messages with code blocks in this format will get
syntax highlighting:

    ```lang
    some.code("here")
    ```

This was achieved by allowing the `code` markdown feature for
chat and also adding a highlight function to the `decorateMessages`
flow in chat-live-pane.

image

### CODE REVIEW

File: assets/javascripts/discourse/components/chat-live-pane.js [hunk 1]
```diff
@@ -1272,6 +1277,24 @@ export default Component.extend({
...
+    document
+      .querySelector(
+        `.chat-message-container-${chatMessageId} .chat-message-text`
+      )
+      ?.classList.remove("hljs-complete");
+  },
+
+  _highlightCode() {
+    document.querySelectorAll(".chat-message-text").forEach((chatMessageEl) => {
+      // no need to do this for every single message every time a message changes
```
Comment (nattsw) [inline assets/javascripts/discourse/components/chat-live-pane.js:1290]: 👍

File: test/javascripts/acceptance/chat-test.js [hunk 2]
```diff
@@ -225,6 +226,8 @@ acceptance("Discourse Chat - without unread", function (needs) {
     Object.defineProperty(this, "appEvents", {
       get: () => this.container.lookup("service:appEvents"),
     });
+    Session.current().highlightJsPath =
+      "/assets/highlightjs/highlight-test-bundle.min.js";
```
Comment (nattsw) [inline test/javascripts/acceptance/chat-test.js:230]: I guess this is needed to get highlighting working in tests? Is this related to the usage of `import highlightSyntax from "discourse/lib/highlight-syntax";` in the implementation?

File: test/javascripts/acceptance/chat-test.js [hunk 3]
```diff
@@ -595,6 +598,48 @@ acceptance("Discourse Chat - without unread", function (needs) {
...
+      stagedId: 1,
+      chat_message: {
+        id: 202,
+        cooked: `<pre><code class="lang-ruby">Widget.triangulate(arg: "test")
+      </code></pre>`,
+        user: {
+          id: 1,
+        },
+      },
+    });
```
Comment (nattsw) [inline test/javascripts/acceptance/chat-test.js:624]: Interesting way of testing 👍

**Review by nattsw — APPROVED** [review]

File: test/javascripts/acceptance/chat-test.js (hunk 2 above)
Comment (martin-brennan) [reply test/javascripts/acceptance/chat-test.js:230]: Yeah I just copied what another test in core was doing with it, `highlight-syntax` loads this library

**Pull request merged**